.github/workflows/rust.yml

```diff
@@ -84,3 +84,35 @@ jobs:
 
       - name: "cargo test"
         run: cd geopolars && cargo test --all --all-features
+
+  benchmark:
+    name: Benchmark
+    runs-on: ubuntu-latest
+    needs: [test, check]
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Install Rust
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          profile: minimal
+          override: true
+
+      - uses: Swatinem/rust-cache@v1
+
+      - name: "cargo bench"
+        run: cd geopolars && cargo bench | tee output.txt
+
+      - name: Store benchmark result
+        uses: benchmark-action/github-action-benchmark@v1
+        with:
+          name: Rust Benchmark
+          tool: 'cargo'
+          output-file-path: output.txt
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          auto-push: true
```

Comment:
This will fail for now because you don't have a `gh-pages` branch!

Reply:
Note: if you ever plan to host docs in gh-pages branch (e.g mkdocs) you might want to set `benchmark-data-dir-path` (https://github.com/benchmark-action/github-action-benchmark#benchmark-data-dir-path-required) to save the data.js/index.html in a sub directory